### DISCUSSION TITLE
how did you learn how to do this port?

### DISCUSSION THREAD

**Discussion by chetmurthy** [rrevenantt/antlr4rust#69] [General]
I'd like to do an ANTLR port (to OCaml) but haven't actually looked into how to get it done.  I saw you'd done this port, and wondered if you had pointers to where to start learning how to port ANTLR to a new language runtime ?

**Comment by rrevenantt** [comment]
Well its not like you learn that somewhere. In general if there is already a target for similar language then porting ANTLR runtime is a quite straightforward rewriting task. The most important similarity is garbage collection and OOP/inheritance. For Rust there was no both no-GC no-inheritance target before(and i also wanted to make it faster and safer than C++ one) so it was quite non-trivial task. But OCaml seems to be both GC and OOP language so you shouldn't have any complications porting directly from Java, maybe even will be able to use some automatic conversion tool as a first step.
